### DISCUSSION TITLE
Persist into HASPmota

### DISCUSSION THREAD

**Discussion by AK1Chel** [arendst/Tasmota#21330] [Q&A] [answered]
I use this line to insert mem1 data to HASPmota.
{"page":1,"id":12,"obj":"label","x":20,"y":115,"h":30,"w":220,"align":1,"text":"---","text_rule":"mem1","text_rule_format":" %s","text_font":"Futura.ttf-28","text_color":"#FFFF00"}

How to insert persist.value?

**Comment by sfromis** [comment]
Are you asking about data stored with the `persist` module of Berry?

In that case, I'd expect a construct like this `"text_rule_formula":"persist.myvalue"` to pick up the value, but then you have to decide when to look for values, maybe in a timer at intervals.

However, it may be simpler to do something like `p1b12.text=persist.myvalue` (or something along those lines) when you are updating the `persist` value in Berry.

**Comment by AK1Chel** [comment]
Thanks for the answer.
I use this method when changing persist.myvalue.
But when turned on it doesn't work. It is necessary to initiate each value at startup. For RULE variables it works fine.

**Reply by sfromis** [reply, marked as answer]
At startup, you can have the `p1b12.text=persist.myvalue` in `autoexec.be` or something like that. I suppose that you could also do it in the `.jsonl` file by triggering on e.g. `System#boot`.

If you have reasons to prefer doing it by command, you can can use the `Br` command to execute a tiny bit of Berry code to return a value:
`Br import persist return persist.myvalue`

**Reply by AK1Chel** [reply]
Thank you
